core/model-vocabulary/src/main/java/org/eclipse/rdf4j/model/vocabulary/XSD.java

```diff
@@ -186,58 +189,58 @@
-		GMONTH(XSD.GMONTH, true, false, false, false, false, false, true),
-		STRING(XSD.STRING, true, false, false, false, false, false, false),
-		BOOLEAN(XSD.BOOLEAN, true, false, false, false, false, false, false),
-		BASE64BINARY(XSD.BASE64BINARY, true, false, false, false, false, false, false),
-		HEXBINARY(XSD.HEXBINARY, true, false, false, false, false, false, false),
-		FLOAT(XSD.FLOAT, true, false, false, false, false, true, false),
-		DECIMAL(XSD.DECIMAL, true, false, false, false, true, false, false),
-		DOUBLE(XSD.DOUBLE, true, false, false, false, false, true, false),
-		ANYURI(XSD.ANYURI, true, false, false, false, false, false, false),
-		QNAME(XSD.QNAME, true, false, false, false, false, false, false),
-		NOTATION(XSD.NOTATION, true, false, false, false, false, false, false),
-		NORMALIZEDSTRING(XSD.NORMALIZEDSTRING, false, false, false, true, false, false, false),
-		TOKEN(XSD.TOKEN, false, false, false, true, false, false, false),
-		LANGUAGE(XSD.LANGUAGE, false, false, false, true, false, false, false),
-		NMTOKEN(XSD.NMTOKEN, false, false, false, true, false, false, false),
-		NMTOKENS(XSD.NMTOKENS, false, false, false, true, false, false, false),
-		NAME(XSD.NAME, false, false, false, true, false, false, false),
-		NCNAME(XSD.NCNAME, false, false, false, true, false, false, false),
-		ID(XSD.ID, false, false, false, true, false, false, false),
-		IDREF(XSD.IDREF, false, false, false, true, false, false, false),
-		IDREFS(XSD.IDREFS, false, false, false, true, false, false, false),
-		ENTITY(XSD.ENTITY, false, false, false, true, false, false, false),
-		ENTITIES(XSD.ENTITIES, false, false, false, true, false, false, false),
-		INTEGER(XSD.INTEGER, false, false, true, true, true, false, false),
-		LONG(XSD.LONG, false, false, true, true, true, false, false),
-		INT(XSD.INT, false, false, true, true, true, false, false),
-		SHORT(XSD.SHORT, false, false, true, true, true, false, false),
-		BYTE(XSD.BYTE, false, false, true, true, true, false, false),
-		NON_POSITIVE_INTEGER(XSD.NON_POSITIVE_INTEGER, false, false, true, true, true, false, false),
-		NEGATIVE_INTEGER(XSD.NEGATIVE_INTEGER, false, false, true, true, true, false, false),
-		NON_NEGATIVE_INTEGER(XSD.NON_NEGATIVE_INTEGER, false, false, true, true, true, false, false),
-		POSITIVE_INTEGER(XSD.POSITIVE_INTEGER, false, false, true, true, true, false, false),
-		UNSIGNED_LONG(XSD.UNSIGNED_LONG, false, false, true, true, true, false, false),
-		UNSIGNED_INT(XSD.UNSIGNED_INT, false, false, true, true, true, false, false),
-		UNSIGNED_SHORT(XSD.UNSIGNED_SHORT, false, false, true, true, true, false, false),
-		UNSIGNED_BYTE(XSD.UNSIGNED_BYTE, false, false, true, true, true, false, false),
-		YEARMONTHDURATION(XSD.YEARMONTHDURATION, false, true, false, true, false, false, false);
+		DURATION(create("duration"), true, true, false, false, false, false, false),
+		DATETIME(create("dateTime"), true, false, false, false, false, false, true),
```

Comment:
I'm not sure I follow - it looks like you are now creating IRI constants for the XSD datatypes both here _and_ in the inner enum in `CoreDatatype`. Shouldn't one reuse the other?

Reply:
Good point. The XSD vocabulary class should probably use the IRI constants from the CoreDatatype.XSD enums.